v2/router.go

```diff
@@ -87,8 +87,11 @@ type EncodingProcedure struct {
 	// Human-readable signature of the handler.
 	Signature string
 
-	// Codec to assist mapping the encoding-level handler to transport-level handler
+	// Codec to assist mapping bodies between the encoding-level handler to transport-level handler
 	Codec func() InboundCodec
+
+	// Codec to assist mapping errors between the encoding-level handler to transport-level handler
+	ErrorCodec func() InboundErrorCodec
```

Comment:
May I propose that ErrorCodec be merged into Codec?

Reply:
Seconded. This would ease the "statefulness" issue, since we need to rely on some information from decoding in Thrift and Protobuf's case.